zerver/webhooks/opensearch/view.py

```diff
@@ -0,0 +1,33 @@
+from typing import Annotated
+
+from django.http import HttpRequest, HttpResponse
+
+from zerver.decorator import webhook_view
+from zerver.lib.response import json_success
+from zerver.lib.typed_endpoint import ApiParamConfig, typed_endpoint
+from zerver.lib.webhooks.common import check_send_webhook_message
+from zerver.models import UserProfile
+
+
+@webhook_view("OpenSearch")
+@typed_endpoint
+def api_opensearch_webhook(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    *,
+    payload: Annotated[str, ApiParamConfig(argument_type_is_body=True)],
+) -> HttpResponse:
+    """
+    OpenSearch only sends text/plain payloads, even when the Content-Type is
+    set to other formats.
```

Comment:
Is there documentation we can link to here?

Reply:
Yeah, this seems OK if there's not a good resource.